packages/metro-react-native-babel-preset/src/configs/main.js

```diff
@@ -199,13 +199,17 @@ const getPreset = (src, options) => {
         plugins: [
           [
             require('@babel/plugin-transform-typescript'),
-            {
-              isTSX: true,
-              allowNamespaces: true,
-            },
+            optionsForTransformTS({isTSX: true}),
           ],
         ],
       },
+      {
+        plugins: [
+          ...defaultPluginsBeforeRegenerator,
+          isHermes ? null : require('@babel/plugin-transform-regenerator'),
+          ...defaultPluginsAfterRegenerator,
+        ].filter(Boolean),
+      },
```

Comment:
Can you elaborate on why this needed to move below the TypeScript passes?

Reply:
Yes, I think it's for a similar reason that the comment above the flow plugin mentions. If the TS transforms are not first, the tests fail with the following error:
```
SyntaxError: TypeScript.ts: /Users/andrew/Development/Libraries/metro/TypeScript.ts: TypeScript 'declare' fields must first be transformed by @babel/plugin-transform-typescript.
    If you have already enabled that plugin (or '@babel/preset-typescript'), make sure that it runs before any plugin related to additional class features:
     - @babel/plugin-proposal-class-properties
     - @babel/plugin-proposal-private-methods
     - @babel/plugin-proposal-decorators
```
It seemed like a good change to have the metro preset do the type removal properties first, for both Flow and TypeScript, so that the overall behavior is consistent between .js and .ts files.